File: Master/1st Course/Spring semester/Moderno programmesanas valodu praktikums/Lab_8/Company.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace Lab8 {

    public class Company : ICompanyService {

        public string Title { get; set; }
        private List<Employee> employers = new List<Employee>();

        public Company(string title) {
            Title = title;
        }

        public void Add(Employee employee) {
            employers.Add(employee);
        }

        public double FindMaxSalary() {
            double? maxSalary = 
                employers.OrderByDescending(e => e.Salary).Select(e => e.Salary).FirstOrDefault();

            return maxSalary ?? 0;
        }

        internal class SalDesc : IComparer<Employee> {

            public int Compare(Employee e1, Employee e2) {
                return e1.Salary == e2.Salary ? 0 : e1.Salary > e2.Salary ? -1 : 1;
            }
        }

        public void SortSalariesAsc() {
            employers.Sort();
        }

        public void SortSalariesDesc() {
            employers.Sort(new SalDesc());
        }

        public override string ToString() {
            var s = string.Format("Firma: {0}.\n", Title);
            int i = 0;

            employers.ForEach(e => s += string.Format("{0}. {1}\n", ++i, e.ToString()));

            return s;
        }
    }
}

File: Master/1st Course/Spring semester/Moderno programmesanas valodu praktikums/Lab_2/Company.java
package lab2.concepts;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;

public class Company implements CompanyService {
    private String title;
    private ArrayList<Employee> employees = new ArrayList<Employee>();
    
    public Company(String name) {
        this.title = name;
    }
    
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
    
    public void add(Employee employee) {
        employees.add(employee);
    }

    public double findMaxSalary() {
        double maxSalary = 0; 
            
        for (Employee e : employees)
            if (e.getSalary() > maxSalary)
                maxSalary = e.getSalary();

        return maxSalary;
    }
    
    public class SalDesc implements Comparator<Employee> {
        @Override
        public int compare(Employee e0, Employee e1) {
            return e0.getSalary() > e1.getSalary() ? -1 :
                e0.getSalary() < e1.getSalary() ? 1 : 0;
        }
    }
    
    public void sortSalariesAsc() {
        Collections.sort(employees);
    }
    
    public void sortSalariesDesc() {
        Collections.sort(employees, new SalDesc());
    }

    @Override
    public String toString() {
        String s = "Firma: " + title + ".\n";
        
        for (int i = 0; i < employees.size(); i++)
            s += (i + 1) + ". " + employees.get(i).toString() + "\n";
        
        return s;
    }
}
